cmd/object-utils.go

```diff
@@ -163,28 +163,27 @@ func (d byBucketName) Len() int           { return len(d) }
 func (d byBucketName) Swap(i, j int)      { d[i], d[j] = d[j], d[i] }
 func (d byBucketName) Less(i, j int) bool { return d[i].Name < d[j].Name }
 
-// Copied from io.LimitReader()
-// limitReader returns a Reader that reads from r
-// but returns error after n bytes.
-// The underlying implementation is a *LimitedReader.
-type limitedReader struct {
-	R io.Reader // underlying reader
-	M int64     // min bytes remaining
-	N int64     // max bytes remaining
+// rangeReader returns a Reader that reads from r
+// but returns error after Max bytes read as errDataTooLarge.
+// but returns error if reader exits before reading Min bytes
+// errDataTooSmall.
+type rangeReader struct {
+	bucket, object string
```

Comment:
bucket, object here is not needed.

Because it is added (rightly) here:

```
                bytesWritten, err = fsCreateFile(fs.storage, teeReader, buf, minioMetaTmpBucket, tempOb\
j)                                                                                                      
                if err != nil {                                                                         
                        fs.storage.DeleteFile(minioMetaTmpBucket, tempObj)                              
                        errorIf(err, "Failed to create object %s/%s", bucket, object)                   
                        return ObjectInfo{}, toObjectErr(err, bucket, object)                           
                }
```

rangeReader can be a primitive without knowledge of bucket/object